pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/conditions/NotBuilt.groovy

```diff
@@ -46,7 +46,7 @@ class NotBuilt extends BuildCondition {
 
     @Override
     boolean meetsCondition(@Nonnull WorkflowRun r, Object context, Throwable error) {
-        return combineResults(r, error) == Result.NOT_BUILT
+        return combineResults(r, error, null, null) == Result.NOT_BUILT
```

Comment:
Someone could use NOT_BUILT inside of catchError (although it would probably not make much sense). My vote would be to make this match all of the other specific-result conditions.

Reply:
Good point.